src/main/java/org/opensearch/sdk/ExtensionApi.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.sdk;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class encapsulates the API of an Extension.
+ */
+public class ExtensionApi {
+
+    private List<String> api = new ArrayList<>();
+
+    /**
+     * Placeholder field. Change the location to extension_api.yml file of the extension.
+     */
+    public static final String EXTENSION_API_DESCRIPTOR = "src/test/resources/extension_api.yml";
```

Comment:
This is nice. Can we generate these RestAPI configuration from a Spec[1]? 
Just something for the future. 
We'd like to have extensions defining RestAPIs and having our clients to support it. 
cc: @VachaShah

[1] https://github.com/opensearch-project/OpenSearch/issues/3090

Reply:
+1 on spec-based OpenAPI-compatible generation.

Reply:
See also the comments at the header of that file:
```
# This is a placeholder for a more formal API definition file, e.g., an OpenAPI
# spec file which can be used to generate documentation and streamline testing
# See https://github.com/opensearch-project/OpenSearch/issues/3090
# The API will include at least two components, a method (e.g., GET, PUT) and
# parts of a URI (e.g., /_extensions/sample-extension/api_1)
```